main.go

```diff
@@ -251,3 +328,21 @@ func rawSetuid(uid int) error {
 
 	return nil
 }
+
+func (handler *BuildHandler) saveNewBuild(repoName string) BuildInfo {
+	info := BuildInfo{
+		repository: repoName,
+		startedAt:  time.Now(),
+		status:     "in progress",
+
+		pointer: handler.currentBuild,
```

Comment:
pointer needed for updating build in ring buffer (after build compleat). I can move it out of `BuildInfo` structure, buth then it should be returned as second return argument.